game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -292,7 +292,7 @@ public static void showMainFrame() {
   private static void loadGame() {
     try {
       newBackgroundTaskRunner().runInBackground("Loading game...", () -> {
-        gameSelectorModel.loadDefaultGame(false);
+        gameSelectorModel.loadDefaultGame();
```

Comment:
The reason the progress dialog disappears so quickly is because `GameSelectorModel#loadDefaultGame()` spawns a new thread to do its work, whereas the legacy `loadDefaultGame(boolean)` overload did not.  So we're effectively spawning two threads to load the game at client startup.

The obvious fix to correct the immediate problem is to not spawn a new thread in `loadDefaultGame()`.  However, `loadDefaultGame()` is called in multiple locations, and they may not behave correctly if blocked while loading the game.

I seem to remember @RoiEXLab and I having this discussion before about the blocking vs. non-blocking `loadDefaultGame()` overloads to make the UI more friendly.  We may have just settled on having the two overloads for that reason.

Reply:
Making `GameSelectorModel#loadDefaultGameTask()` public and calling it on the above line instead of `loadDefaultGame()` also fixes the problem, but the methods may need to be renamed to avoid the smell (e.g. `loadDefaultGameInBackground()` vs. `loadDefaultGame()`).

Reply:
Good catch, thank you. This should be fixed in: b14b570; in that commit I made the newThread sameThread behavior explicit.

Reply:
Yes, the new names are much clearer. :+1: